scipy/optimize/_minimize.py

```diff
@@ -202,6 +204,17 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
     and either the Hessian or a function that computes the product of
     the Hessian with a given vector.
 
+    Method :ref:`trust-iterative <optimize.minimize-trustiterative>`
+    uses a iterative trust-region algorithm described in [13]_ for
+    unconstrained minimization.  This  algorithm requires
+    the gradient and Hessian; Recomended for medium size problems;
+    The advantage of this method is that it can provide a much more
+    accurate solution for the trust-region subproblem than, for instance,
+    the dogleg method.  For each iteration it solve the trust-region
+    subproblem iteratively requiring typically three Cholesky
+    factorizations (agaist the single Cholesky factorization needed by dogleg
+    method).
+
```

Comment:
I am working on some examples of this! I pretend to include it on the my next commit...